src/commands/lambda/__tests__/functions/instrument.test.ts

```diff
@@ -458,7 +458,181 @@ describe('instrument', () => {
+          process.env[CI_KMS_API_KEY_ENV_VAR] = '5678'
+          const badSettings = {...settings, extensionVersion: 24, layerVersion: 4}
+          let error
+          try {
+            await calculateUpdateRequest(javaConfig, badSettings, region, javaRuntime)
+          } catch (e) {
+            if (e instanceof Error) {
+              error = e
+            }
+          }
+          expect(error?.message).toBe(
+            `For the ${javaRuntime} runtime, the dd-trace version 4 is not compatible with the dd-extension version 24`
+          )
+        })
+
+        const baseVariables = {
+          DD_KMS_API_KEY: '5678',
+          DD_MERGE_XRAY_TRACES: 'false',
+          DD_SITE: 'datadoghq.com',
+          DD_TRACE_ENABLED: 'false',
+        }
+        const compatibleTradeAndExtension = {
+          Environment: {
+            Variables: {...baseVariables, AWS_LAMBDA_EXEC_WRAPPER: '/opt/datadog_wrapper'},
+          },
+          FunctionName: 'arn:aws:lambda:us-east-1:123456789012:function:lambda-hello-world',
+          Layers: [
+            'arn:aws:lambda:us-east-1:123456789012:layer:Datadog-Extension:25',
+            'arn:aws:lambda:us-east-1:123456789012:layer:dd-trace-java:5',
+          ],
+        }
+        const oldExtensionVersion = {
+          Environment: {
+            Variables: {
+              ...baseVariables,
+            },
+          },
+          FunctionName: 'arn:aws:lambda:us-east-1:123456789012:function:lambda-hello-world',
+          Layers: [
+            'arn:aws:lambda:us-east-1:123456789012:layer:Datadog-Extension:23',
+            'arn:aws:lambda:us-east-1:123456789012:layer:dd-trace-java:2',
+          ],
+        }
+        const traceUndefined = {
+          Environment: {
+            Variables: {
+              ...baseVariables,
+            },
+          },
+          FunctionName: 'arn:aws:lambda:us-east-1:123456789012:function:lambda-hello-world',
+          Layers: ['arn:aws:lambda:us-east-1:123456789012:layer:Datadog-Extension:15'],
+        }
+
+        test.each`
+          extensionVersion | traceVersion | outputResult
+          ${25}            | ${5}         | ${compatibleTradeAndExtension}
+          ${23}            | ${2}         | ${oldExtensionVersion}
+          ${15}            | ${undefined} | ${traceUndefined}
+        `(
+          'should the output match the expected if extensionVersion=$extensionVersion and traceVersion=$traceVersion',
+          async ({extensionVersion, traceVersion, outputResult}) => {
+            const curSettings = {...settings, extensionVersion, layerVersion: traceVersion}
+            process.env[CI_KMS_API_KEY_ENV_VAR] = '5678'
+            const updateRequest = await calculateUpdateRequest(javaConfig, curSettings, region, javaRuntime)
+            expect(updateRequest).toEqual(outputResult)
+          }
+        )
+      })
+    })
   })
+
```

Comment:
This file is now > 1000 lines. Might make sense to split the tests up into a few files for readability.

Reply:
Totally agree. Some kind of refactoring is also necessary now.